lib/ShlinkProxy.php

```diff
@@ -0,0 +1,84 @@
+<?php declare(strict_types=1);
+/**
+ * PrivateBin
+ *
+ * a zero-knowledge paste bin
+ *
+ * @link      https://github.com/PrivateBin/PrivateBin
+ * @copyright 2012 Sébastien SAUVAGE (sebsauvage.net)
+ * @license   https://www.opensource.org/licenses/zlib-license.php The zlib/libpng License
+ */
+
+namespace PrivateBin;
+
+/**
+ * ShlinkProxy
+ *
+ * Forwards a URL for shortening to shlink and stores the result.
+ */
+class ShlinkProxy extends AbstractProxy
+{
+    /**
+     * constructor
+     *
+     * initializes and runs ShlinkProxy
+     *
+     * @access public
+     * @param string $link
+     */
+    public function __construct(Configuration $conf, $link)
+    {
+        parent::__construct($conf, $link);
+    }
+
+    /**
+     * Overrides the abstract parent function to get contents from Shlink API.
+     *
+     * @access protected
+     * @return string
+     */
+    protected function _getcontents(Configuration $conf, string $link)
+    {
+        $shlink_api_url = $conf->getKey('apiurl', 'shlink');
+        $shlink_api_key = $conf->getKey('apikey', 'shlink');
+
+        if (empty($shlink_api_url) || empty($shlink_api_key)) {
+            return;
+        }
+
+        $body = array(
+            'longUrl' => $link,
+        );
+
+        return file_get_contents(
+            $shlink_api_url, false, stream_context_create(
+                array(
+                    'http' => array(
+                        'method'  => 'POST',
+                        'header'  => "Content-Type: application/json\r\n" .
+                                     'X-Api-Key: ' . $shlink_api_key . "\r\n",
+                        'content' => Json::encode($body),
+                    ),
+                )
+            )
+        );
+    }
+
+    /**
+     * Extracts the short URL from the shlink API response.
+     *
+     * @access protected
+     * @param array $data
+     * @return ?string
+     */
+    protected function _extractShortUrl(array $data): ?string
+    {
+        if (
+            !is_null($data) &&
```

Comment:
As [Scrutinizer points out](https://scrutinizer-ci.com/g/PrivateBin/PrivateBin/inspections/5259d197-a9b9-467d-b00e-1925aec8c386/issues/files/lib/ShlinkProxy.php?status=new&orderField=path&order=asc&honorSelectedPaths=0), because we now enforce $data to be an array in the method signature, this check can no longer fail.
```suggestion
```